lib/lib-table-query/src/main/java/org/sagebionetworks/table/query/util/FacetUtils.java

```diff
@@ -15,39 +22,157 @@ public class FacetUtils {
 		ValidateArgument.required(validatedFacets, "validatedFacets");
 		
 		if(validatedFacets.isEmpty()){
 			return null;
 		}
-		
+
+		List<FacetRequestColumnModel> facetColumnsWithoutIgnoredColumn = validatedFacets.stream()
+				.filter(facet -> {
+					if (columNameToIgnore.isPresent()) {
+						return !facet.getColumnName().equals(columNameToIgnore.get());
+					} else {
+						return true;
+					}
+				})
+				.collect(Collectors.toList());
+
+		List<String> conditionStrings = facetColumnsWithoutIgnoredColumn.stream().map(FacetRequestColumnModel::getSearchConditionString).collect(Collectors.toList());
+
+		return concatConditionStrings(conditionStrings);
+	}
+
+
+	/**
+	 * Concatenates a list of query filter Strings with AND and then wraps that search condition with a parenthesis
+	 * e.g. ["(col1 LIKE 'abc' OR col1 LIKE '%xyz')", "(col2 HAS('d'))"] => "( (col1 LIKE 'abc' OR col1 LIKE '%xyz') AND (col2 HAS('d')) )"
+	 * @return the concatenated string or null if there was nothing to concatenate
+	 */
+	public static String concatQueryFilterConditionStrings(List<QueryFilter> filters){
+		if(filters == null || filters.isEmpty()){
+			return null;
+		}
+
+		return concatConditionStrings(filters.stream().map(FacetUtils::getSearchConditionString).collect(Collectors.toList()));
+	}
+
+	/**
+	 * Concatenates a list of SQL conditional statements with " AND "
+	 * @param conditionStrings
+	 * @return
+	 */
+	private static String concatConditionStrings(List<String> conditionStrings) {
+		ValidateArgument.required(conditionStrings, "conditionStrings");
+
 		StringBuilder builder = new StringBuilder();
 		builder.append("(");
 		int initialSize = builder.length(); //length with the "( " included
-		
-		for(FacetRequestColumnModel facetColumn : validatedFacets){
-			if(columNameToIgnore == null || !facetColumn.getColumnName().equals(columNameToIgnore)){ //make sure not to include the ignored column
-				String searchCondition = facetColumn.getSearchConditionString();
-				if(searchCondition != null){//don't add anything if there is no search condition
-					if(builder.length() > initialSize){ //not the first element
-						builder.append(" AND ");
-					}
-					builder.append(searchCondition);
-				}
-			}
-		}
+
+		String joinedConditions = conditionStrings.stream()
+				// Remove nulls
+				.filter(s -> s != null && !s.isEmpty())
+				// Join with " AND "
+				.collect(Collectors.joining(" AND "));
+
+		builder.append(joinedConditions);
+
 		if(builder.length() == initialSize){ //edge case where nothing got concatenated together
 			return null;
 		}
-		
+
+		builder.append(")");
+		return builder.toString();
+	}
+
+	/**
+	 * Converts a QueryFilter to a Synapse SQL condition string
+	 * @param filter
+	 * @return a standalone condition, such as "( \"foo\" LIKE '%bar%' )"
+	 */
+	public static String getSearchConditionString(QueryFilter filter) {
+		if (filter instanceof ColumnSingleValueQueryFilter) {

```

Comment:
Find a way to move this logic out of a static util and into a PoJo with methods like toSql().